geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -58,6 +62,33 @@ class AccumuloDataStore(val connector: Connector, override val config: AccumuloD
   private val statsTable = GeoMesaFeatureIndex.formatSharedTableName(config.catalog, "stats")
   override val stats = new AccumuloGeoMesaStats(this, statsTable, config.generateStats)
 
+  // If on a secured cluster, create a thread to periodically renew Kerberos tgt
+  private val kerberosRenewerThread : Option[Thread] = if (UserGroupInformation.isSecurityEnabled) {
+    logger.info("Creating thread to periodically renew TGT from keytab")
+    Some(new Thread(new Runnable {
+      def run() {
+        while (true) {
+          try {
+            logger.info("Checking whether TGT needs renewing...")
+            UserGroupInformation.getCurrentUser.checkTGTAndReloginFromKeytab()
+          } catch {
+            case iox: IOException => logger.warn("Error checking and renewing tgt: " + iox.toString)
+          }
+
+          try {
+            // Sleep for 10 mins
+            Thread.sleep(10*60*1000)
+          } catch {
+            case ie: InterruptedException => logger.info("Kerberos renewer thread stopping"); return
```

Comment:
Instead of tracking sleeping, running, etc in your thread, you could use a scheduled executor service to run it repeatedly: `val exector = Executors.newSingleThreadScheduledExecutor(); executor.scheduleAtFixedRate(...)`
you'd want to shut it down in the dispose method.
Using an executor is generally preferred over `Thread.start` anyway, although for a simple use case like this it probably doesn't matter too much